File: Daily Temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures)
    {
        stack <pair<int,int>> temp;
        int n=temperatures.size(); 
        vector <int> ans(n);

        for(int i=0;i<n;i++)
        {
            while(!temp.empty() && temp.top().first <temperatures.at(i))
            {
                ans.at(temp.top().second)=i-temp.top().second;
                temp.pop();
            }
            
            temp.push({temperatures.at(i),i});
        }
            while(!temp.empty())
            {
                ans.at(temp.top().second)=0;
                temp.pop();
            }
            return ans;
    }
      
};

File: pre-camp/week/leetcode/daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        stack = []
        n = len(temperatures)
        ans = [0] * n
        
        for i in range(n):
            while stack and temperatures[stack[-1]] < temperatures[i]:
                index = stack.pop()
                ans[index] = i - index
            stack.append(i)
        
        return ans